### PR TITLE
fix: custom css with New Teams

### LINKED ISSUE #1013: Custom CSS does not work for Teams v2
**Describe the bug**
The --customCssLocation loading does not work for the new Teams version.

**To Reproduce**
1. Start teams with `--customCSSLocation` with some css
2. The custom CSS that works in Edge with the Stylus extension does nothing in teams-on-linux.

**Expected behavior**
Custom CSS is applied.

**Desktop (please complete the following information):**
 - OS: Linux
 - Installation package: deb
 - Version 1.3.19

**Additional context**
The problem is with v2 only, and I have updated my css for v2 teams - the problem seems to be that v2 teams use `iframe`, and the `webContents.insertCss` does not work for it.

I have tried to call the insertCSS for frame's webContents in various ways but none if it worked, so I have done this on my local PC, but it's rather ugly and I assume that you don't want this as PR:

```diff
diff --git a/app/mainAppWindow/index.js b/app/mainAppWindow/index.js
index 40447dd..67a0127 100644
--- a/app/mainAppWindow/index.js
+++ b/app/mainAppWindow/index.js
@@ -1,5 +1,16 @@
 require('@electron/remote/main').initialize();
-const { shell, BrowserWindow, ipcMain, app, session, nativeTheme, powerSaveBlocker, dialog } = require('electron');
+const {
+  shell,
+  BrowserWindow,
+  ipcMain,
+  app,
+  session,
+  nativeTheme,
+  powerSaveBlocker,
+  dialog,
+  webContents,
+  webFrameMain,
+} = require("electron");
 const isDarkMode = nativeTheme.shouldUseDarkColors;
 const windowStateKeeper = require('electron-window-state');
 const path = require('path');
@@ -298,6 +309,24 @@ function addEventHandlers() {
        window.webContents.session.webRequest.onBeforeSendHeaders(getWebRequestFilterFromURL(), onBeforeSendHeadersHandler);
        login.handleLoginDialogTry(window);
        window.webContents.on('did-finish-load', onDidFinishLoad);
+
+       window.webContents.on("did-frame-finish-load", (event, isMain, frameProcessId, frameRoutingId) => {
+               const wf = webFrameMain.fromId(frameProcessId, frameRoutingId);
+
+               const fs = require("fs");
+
+               fs.readFile(config.customCSSLocation, "utf-8", (error, data) => {
+                       if (!error) {
+                               wf.executeJavaScript(`
+                                       var style = document.createElement('style');
+                                       style.innerHTML = \u0060${data}\u0060;
+                                       document.head.appendChild(style);
+                               `);
+                       }
+               });
+       });
        window.on('closed', onWindowClosed);
        window.webContents.addListener('before-input-event', onBeforeInput);
 }
```

### PR DESCRIPTION
**Pull request by Tasssadar**
New Teams use iframe for the main content view, and `insertCSS` does not weem to work for that. Use ugly hacks to fix it.

Fixes #1013

### CODE REVIEW

File: app/mainAppWindow/index.js [hunk 1]
```diff
@@ -155,6 +155,17 @@ function onDidFinishLoad() {
 	customCSS.onDidFinishLoad(window.webContents, config);
 }
 
+function onDidFrameFinishLoad(event, isMainFrame, frameProcessId, frameRoutingId) {
+	logger.debug('did-frame-finish-load', event, isMainFrame);
+
+	if (isMainFrame) {
+		return; // We want to insert CSS only into the Teams V2 content iframe
+	}
```
Comment (Tasssadar) [inline app/mainAppWindow/index.js:163]: I _think_ this should ensure we don't break the old Teams with this, but I can't really test that easily.

File: app/customCSS/index.js [hunk 2]
```diff
@@ -2,23 +2,60 @@ const fs = require('fs');
...
+
+		webFrame.executeJavaScript(`
+			if(!document.getElementById("${customCssId}")) {
+				var style = document.createElement('style');
+				style.id = "${customCssId}";
+				style.type = "text/css";
+				style.innerHTML = \u0060${data}\u0060;
+				document.head.appendChild(style);
+			}
+		`);
```
Comment (Tasssadar) [inline app/customCSS/index.js:59]: There is IMHO still some potential for JS injection here, but I'd guess we don't care since it is user supplied anyway.

**Comment by IsmaelMartinez** [comment]
Thanks for fixing this. Can you increase the version number in the package.json (just the x in the 1.3.x) and then I would be able to release it. 

Thanks again!

**Comment by Tasssadar** [comment]
Added, also updated some changelog there.

**Comment by IsmaelMartinez** [comment]
released in https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v1.3.20 Thanks again!

**Comment by IsmaelMartinez** [comment]
Hi @dserv-nh , that depends on your organisation. Some people/organisations are getting a toggle option, while others are just getting the new version altogether. Nothing we can do from our side.

**Pull request merged**